vignettes/usermod_vignette.Rmd

```diff
@@ -0,0 +1,94 @@
+---
+title: "How to Write a User Modification Chunk"
+date: "`r Sys.Date()`"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{usermod_chunks}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+library(devtools)
+devtools::load_all()
+```
+
+## Introduction
+Users may want to change the default gcamdata behavior by either modifying input assumptions or changing intermediate chunks. They can now write a "user modification" chunk that can be "plugged in" to the data system. This new chunk can modify any objects that are used or created in gcamdata and pass the modified object to all dependent chunks.
+
+User-modification chunks have a format similar to other data chunks in gcamdata, except that instead of producing a new output, it returns a modified data object that replaces the original data object in the data system. These new chunks can be added to `driver_drake()` or `driver()` using the new arguments `user_modifications` and `user_suffix`, which tell gcamdata which modification function to use and what suffix to add to all impacted downstream objects and xmls. 
+
+## Example: Modify Shareweight
+Below we show an example user-modification chunk to change a shareweight in an input csv file.
+
+### User Modification Chunk
+Here we load in two csv files, "energy/A322.subsector_shrwt.csv" and "common/GCAM_region_names.csv". We modify A322.subsector_shrwt, so we list it under `driver.DECLARE_MODIFY`, but do not modify GCAM_region_names, so it is listed under `driver.DECLARE_INPUTS`. Then, we set the shareweight column of the first row of A322.subsector_shrwt to `NEW.SHWT`. Finally, we use a new `return_modified()` function to return the modified A322.subsector_shrwt (note that we have to include the path for input files).
```

Comment:
maybe better described as "note that we have to match the original object name we asked for in `driver.DECLARE_MODIFY`, which for inputs files this means including the file path"